In [26]:
import pandas as pd
from pandas import Series, DataFrame

df_2019_jul = pd.read_csv('../data/nyc_taxi_2019-07.csv',
                usecols=['passenger_count', 
                        'total_amount', 'payment_type'])
df_2019_jul['year'] = 2019

df_2020_jul = pd.read_csv('../data/nyc_taxi_2020-07.csv',
                usecols=['passenger_count', 
                        'total_amount', 'payment_type'])
df_2020_jul['year'] = 2020

df = pd.concat([df_2019_jul, df_2020_jul])

# Beyond 1

Use the `corr` method on `df` to find the correlations among the columns. How would you interpret these results?

번역: df에 대해 corr 메서드를 사용하여 열들 간의 상관관계를 구하세요. 그 결과를 어떻게 해석하겠습니까?

In [27]:
df.corr()

passenger_count  payment_type  total_amount      year
passenger_count         1.000000      0.016410      0.014943 -0.049558
payment_type            0.016410      1.000000     -0.138561  0.029277
total_amount            0.014943     -0.138561      1.000000 -0.019706
year                   -0.049558      0.029277     -0.019706  1.000000

There is no significant correlation among any of these columns. The closest that we see is between payment type and total amount — but even there, it's an extremely small, negative correlation. 
딱히 이 컬럼들 사이에서 명확한 상관관계는 보여지지 않음. 그나마 큰 상관계수가 -0.138561이다. 하지만 이마저도 극히 작은 값이다.

+ 결제 유형은 범주형인데, 1이 카드, 2가 현금이라면 
양의 상관이면 요금이 높을수록 현금 사용이 늘어난다는 뜻
음의 상관이면 요금이 높을수록 현금결제를 덜 한다는 뜻
실제로 음의 상관이 나온것을 보아 상식과도 부합하지만, 상관계수의 크기가 매우 미약하다.


(The payment type is actually categorical, so saying that there's a numeric correlation is a bit weird in any event. But given that 1 is credit and 2 is cash, a positive correlation would mean that a higher fare would be more likely to use cash. A negative correlation thus means that a higher fare is *less* likely to be paid in cash, which matches our common sense. But again, the correlation is extremely weak.)

# Beyond 2

Show, with a single command, the difference in descriptive statistics for `total_amount` beween 2019 and 2020. Round values to use no more than 2 digits after the decimal point.

번역: 2019년과 2020년 사이의 total_amount 기술 통계(descriptive statistics) 차이를 단일 명령으로 보여주세요. 값은 소수 둘째 자리까지만 반올림하세요.

In [28]:
(
    df.loc[df['year'] == 2020, 'total_amount'].describe().round(2) - 
    df.loc[df['year'] == 2019, 'total_amount'].describe().round(2)
)
# 2020년 데이터 중 total_amount 열을 선택, describe() 메서드로 요약 통계량 계산 후 
# round 메서드로 소수점 둘째 자리까지 반올림
# 위처럼 2019년 데이터도 계산 후 빼기 연산 수행
# 2020년과 2019년의 요약 통계량 차이 계산


count   -5510007.00
mean          -0.98
std           -0.75
min           53.20
25%           -0.50
50%           -0.60
75%           -0.75
max        -4672.45
Name: total_amount, dtype: float64

# Beyond 3

If we assume that zero-passenger trips are for delivering packages, how were those affected during the pandemic? Show the proportion of such trips in 2019 vs. 2020.

번역: 승객 수가 0인 운행이 소포 배송이라고 가정한다면, 팬데믹 동안 그것들은 어떻게 영향을 받았나요? 2019년과 2020년의 해당 운행 비율을 보여주세요.

In [29]:
df.loc[(df['year'] == 2019) & 
       (df['passenger_count'] == 0), 'passenger_count'].count() / df.loc[(df['year']== 2019),'passenger_count'].count()

np.float64(0.018622599363335383)

In [30]:
df.loc[(df['year'] == 2020) & 
       (df['passenger_count'] == 0), 'passenger_count'].count() / df.loc[(df['year']== 2020),'passenger_count'].count()       

np.float64(0.026446482682882185)

~~Turns out that the proportion of zero-person taxi rides shrank as well, from about 1.6% in 2019 to 0.2% in 2020. Honestly, I expected it to be higher during the pandemic!~~


코드가 잘못 작성되어 있어 고친 코드를 실행한 결과
예상한 대로 팬데믹기간에 탑승객 0명 즉 택배 운송비율이 더 증가한 것을 확인할 수 있음